### ISSUE TITLE
(Benign?) Start of day error messages in felix log 

### ISSUE THREAD

**Issue by robbrockbank** [Metaswitch/calico#367]
Acceptable failure "error" logs should be suppressed.

Snippet from felix.log as follows:

2015-04-15 09:22:23,297 [INFO][2] calico.felix.felix 61: Configuration loaded, starting remaining actors...
2015-04-15 09:22:23,298 [INFO][2] calico.felix.felix 90: Starting actors.
2015-04-15 09:22:23,298 [INFO][2] calico.felix.felix 126: Installing global rules.
2015-04-15 09:22:23,427 [ERROR][4] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 2 failed

Input was:
*nat
--delete PREROUTING --jump felix-PREROUTING
--insert PREROUTING --jump felix-PREROUTING
COMMIT

2015-04-15 09:22:23,428 [ERROR][4] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete PREROUTING --jump felix-PREROUTING'
2015-04-15 09:22:23,558 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 2 failed

Input was:
*filter
--delete INPUT --jump felix-INPUT
--insert INPUT --jump felix-INPUT
COMMIT

2015-04-15 09:22:23,558 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete INPUT --jump felix-INPUT'
2015-04-15 09:22:23,570 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,571 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,571 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT', 'felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,575 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
COMMIT

2015-04-15 09:22:23,575 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,575 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT'] failed
2015-04-15 09:22:23,575 [ERROR][5] calico.felix.fiptables 315: Failed to delete chain felix-FROM-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,579 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,579 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-TO-ENDPOINT'
2015-04-15 09:22:23,580 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,580 [ERROR][5] calico.felix.fiptables 315: Failed to delete chain felix-TO-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,585 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 2 failed

Input was:
*filter
--delete FORWARD --jump felix-FORWARD
--insert FORWARD --jump felix-FORWARD
COMMIT

2015-04-15 09:22:23,585 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete FORWARD --jump felix-FORWARD'
2015-04-15 09:22:23,598 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,599 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,599 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT', 'felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,603 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
COMMIT

2015-04-15 09:22:23,603 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,603 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT'] failed
2015-04-15 09:22:23,603 [ERROR][5] calico.felix.fiptables 315: Failed to delete chain felix-FROM-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,607 [ERROR][5] calico.felix.fiptables 434: Failed to run iptables-restore.
Output:

Error:
iptables-restore: line 3 failed

Input was:
*filter
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,607 [ERROR][5] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-TO-ENDPOINT'
2015-04-15 09:22:23,607 [WARNING][5] calico.felix.fiptables 297: Deleting chains ['felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,607 [ERROR][5] calico.felix.fiptables 315: Failed to delete chain felix-TO-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,718 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 2 failed

Input was:
*filter
--delete INPUT --jump felix-INPUT
--insert INPUT --jump felix-INPUT
COMMIT

2015-04-15 09:22:23,718 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete INPUT --jump felix-INPUT'
2015-04-15 09:22:23,729 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,729 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,729 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT', 'felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,733 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
COMMIT

2015-04-15 09:22:23,733 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,733 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT'] failed
2015-04-15 09:22:23,734 [ERROR][6] calico.felix.fiptables 315: Failed to delete chain felix-FROM-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,737 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,738 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-TO-ENDPOINT'
2015-04-15 09:22:23,738 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,738 [ERROR][6] calico.felix.fiptables 315: Failed to delete chain felix-TO-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,743 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 2 failed

Input was:
*filter
--delete FORWARD --jump felix-FORWARD
--insert FORWARD --jump felix-FORWARD
COMMIT

2015-04-15 09:22:23,743 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 2: '--delete FORWARD --jump felix-FORWARD'
2015-04-15 09:22:23,757 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,758 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,758 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT', 'felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,762 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-FROM-ENDPOINT -
--delete-chain felix-FROM-ENDPOINT
COMMIT

2015-04-15 09:22:23,762 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-FROM-ENDPOINT'
2015-04-15 09:22:23,762 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-FROM-ENDPOINT'] failed
2015-04-15 09:22:23,763 [ERROR][6] calico.felix.fiptables 315: Failed to delete chain felix-FROM-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,767 [ERROR][6] calico.felix.fiptables 434: Failed to run ip6tables-restore.
Output:

Error:
ip6tables-restore: line 3 failed

Input was:
*filter
:felix-TO-ENDPOINT -
--delete-chain felix-TO-ENDPOINT
COMMIT

2015-04-15 09:22:23,767 [ERROR][6] calico.felix.fiptables 436: Non-retryable error on line 3: '--delete-chain felix-TO-ENDPOINT'
2015-04-15 09:22:23,767 [WARNING][6] calico.felix.fiptables 297: Deleting chains ['felix-TO-ENDPOINT'] failed
2015-04-15 09:22:23,767 [ERROR][6] calico.felix.fiptables 315: Failed to delete chain felix-TO-ENDPOINT, giving up. Maybe it is still referenced?
2015-04-15 09:22:23,768 [INFO][2] calico.felix.felix 132: Starting polling for interface and etcd updates.
2015-04-15 09:22:23,768 [INFO][2] calico.felix.felix 139: All top-level actors started, waiting on failures...
2015-04-15 09:22:23,769 [INFO][3] calico.felix.fetcd 110: (Re)connecting to etcd...

**Comment by fasaxc** [comment]
I think there are two minor issues here:

* we should suppress the errors for the expected failures
* we shouldn't be trying to delete those chains; the code will get the correct outcome because it will fail to delete the chains but it's wrong to rely on that defensive fallback.

**Comment by Lukasa** [comment]
This is already covered in #363.
